packages/ui/src/components/TanstackTable.tsx

```diff
@@ -417,10 +436,134 @@ export function StudentsTable({
+
+  // Track screen size for aria-hidden
+  const mediaQuery = typeof window !== 'undefined' ? window.matchMedia('(min-width: 768px)') : null;
+  const [isMediumOrLarger, setIsMediumOrLarger] = useState(false);
+
+  useEffect(() => {
+    // TODO: This is a workaround to avoid a hydration mismatch.
+    // eslint-disable-next-line @eslint-react/hooks-extra/no-direct-set-state-in-use-effect
+    setIsMediumOrLarger(mediaQuery?.matches ?? true);
+  }, [mediaQuery]);
+
+  useEffect(() => {
+    const handler = (e: MediaQueryListEvent) => setIsMediumOrLarger(e.matches);
+    mediaQuery?.addEventListener('change', handler);
+    return () => mediaQuery?.removeEventListener('change', handler);
+  }, [mediaQuery]);
+
+  // Focus the search input when Ctrl+F is pressed
+  useEffect(() => {
+    function onKeyDown(event: KeyboardEvent) {
+      if ((event.ctrlKey || event.metaKey) && event.key.toLowerCase() === 'f') {
+        if (searchInputRef.current && searchInputRef.current !== document.activeElement) {
+          searchInputRef.current.focus();
+          event.preventDefault();
+        }
+      }
+    }
+    document.addEventListener('keydown', onKeyDown);
+    return () => document.removeEventListener('keydown', onKeyDown);
+  }, []);
+
+  return (
+    <div class="card d-flex flex-column h-100">
+      <div class="card-header bg-primary text-white">
+        <div class="d-flex align-items-center justify-content-between gap-2">
+          <div>{title}</div>
+          <div class="d-flex gap-2">
+            {headerButtons}
+
+            {downloadButtonOptions && (
+              <TanstackTableDownloadButton table={table} {...downloadButtonOptions} />
+            )}
+          </div>
+        </div>
+      </div>
+      <div class="card-body d-flex flex-column">
+        <div class="d-flex flex-row flex-wrap align-items-center mb-3 gap-2">
+          <div class="flex-grow-1 flex-lg-grow-0 col-xl-6 col-lg-7 d-flex flex-row gap-2">
+            <div class="input-group">
+              <input
+                ref={searchInputRef}
+                type="text"
+                class="form-control"
+                aria-label={globalFilter.placeholder}
+                placeholder={globalFilter.placeholder}
+                value={globalFilter.value}
+                onInput={(e) => {
+                  if (!(e.target instanceof HTMLInputElement)) return;
+                  globalFilter.setValue(e.target.value);
+                }}
+              />
+              <button
+                type="button"
+                class="btn btn-outline-secondary"
+                aria-label="Clear search"
+                title="Clear search"
+                data-bs-toggle="tooltip"
+                onClick={() => globalFilter.setValue('')}
+              >
+                <i class="bi bi-x-circle" aria-hidden="true" />
+              </button>
+            </div>
+            {/* We do this instead of CSS properties for the accessibility checker */}
```

Comment:
I remember the conversation around this, but don't remember what the actual problem was. Duplicates maybe? Consider making this comment more descriptive.